CONTRIBUTING.md

```diff
@@ -0,0 +1,150 @@
+4. Check if the result list matches the expected one
+
+Which can be translated in a spec file to:
+
+```js
+describe('InstantSearch - Search on specific brand and query filtering', () => {
+  it('navigates to the e-commerce demo', async () => {
+    await browser.url('examples/e-commerce/');
+  });
+
+  it('selects "Appliances" category', async () => {
+    await browser.setSelectedHierarchicalMenuItem('Appliances');
+  });
+
+  it('selects "4 & up" rating', async () => {
+    await browser.setRatingMenuValue('4 & up');
+  });
+
+  it('must have the expected results', async () => {
+    const hitsTitles = await browser.getHitsTitles();
+
+    expect(hitsTitles).toEqual([
+      'Nest - Learning Thermostat - 3rd Generation - Stainless Steel',
+      'LG - 1.1 Cu. Ft. Mid-Size Microwave - Stainless-Steel',
+      'Insignia™ - 2.6 Cu. Ft. Compact Refrigerator - Black',
+      'Keurig - K50 Coffeemaker - Black',
+      'iRobot - Roomba 650 Vacuuming Robot - Black',
+      'Shark - Navigator Lift-Away Deluxe Bagless Upright Vacuum - Blue',
+      'LG - 1.5 Cu. Ft. Mid-Size Microwave - Stainless Steel',
+      'Insignia™ - 5.0 Cu. Ft. Chest Freezer - White',
+      'Samsung - activewash 4.8 Cu. Ft. 11-Cycle High-Efficiency Top-Loading Washer - White',
+      'Samsung - 4.8 Cu. Ft. 11-Cycle High-Efficiency Top-Loading Washer - White',
+      'LG - 2.0 Cu. Ft. Full-Size Microwave - Stainless Steel',
+      'Shark - Rotator Professional Lift-Away HEPA Bagless 2-in-1 Upright Vacuum - Red',
+      'LG - 4.5 Cu. Ft. 8-Cycle High-Efficiency Top-Loading Washer - White',
+      'Samsung - 4.2 Cu. Ft. 9-Cycle High-Efficiency Steam Front-Loading Washer - Platinum',
+      'Panasonic - 1.3 Cu. Ft. Mid-Size Microwave - Stainless steel/black/silver',
+      'LG - 2.0 Cu. Ft. Mid-Size Microwave - Black Stainless',
+    ]);
+  });
+});
+```
+
+General guidelines when writing tests:
+
+- Each step should be translated as a `it` function, while this is not mandatory it helps to make the scenario more readable and properly [separate each actions in Sauce Labs reports](https://user-images.githubusercontent.com/13209/62311104-56217d80-b48b-11e9-94dc-3c18b9ddc2af.png).
+- All actions on the browser are asynchronous, so be sure to always `await` them. **Never run multiple asynchronous commands in parallel as it can confuse some browsers (Internet Explorer)**.
+- Use helper functions when possible, for readability but also for maintainability (if one widget is updated, we only have to updates its helpers without touching the tests). [More about Helpers](#helpers).
+- Do not make assertions to know if an action was correctly performed in the browser. If an action fails (trying to click on an non-existing element for example) then WebdriverIO will automatically throw and fail the test, so asserting on it ourselves is redundant.
+- Only assert what you want to see on the page after an action (Is this element displayed? Is the result list correct? etc.)
```

Comment:
Great advice.
However for the particular case of "is the element displayed?" I think it falls in the category of things wdio automatically waits for and throws if KO: 
Another way to say it is:
Instead of `expect(browser.$$('div').length).to.be(1)` use https://webdriver.io/docs/api/element/waitForDisplayed.html